openreview/venue/venue.py

```diff
@@ -1906,3 +1908,177 @@ def mark_as_accepted(venue_group, edge, submission, user_profile, invite_assignm
                                 print(f'no profile active for {tail}')                                             
         
         return True
+
+    def compute_dual_submission_metadata(self, alternate_venue, output_file_path, sparse_value=5, top_percent_cutoff=1, job_id=None, author_overlap_only=False):
+        short_name_a = self.short_name # Column A
+        short_name_b = alternate_venue.short_name # Column B
+        same_venue = self.venue_id == alternate_venue.venue_id
+
+        print(f'Computing similarity: {short_name_a} ↔ {short_name_b}')
+
+        ## Compute/retrieve scores
+
+        if not job_id:
+            res = self.client.request_paper_similarity(
+                name=f'{short_name_a}--{short_name_b}-Paper-Similarity',
+                venue_id=self.get_submission_venue_id(),
+                alternate_venue_id=alternate_venue.get_submission_venue_id(),
+                model='specter2+scincl',
+                sparse_value=sparse_value
+            )
+            job_id = res['jobId']
+            print('Computing scores... Job ID: ', job_id)
+
+        results = self.client.get_expertise_results(job_id=job_id, wait_for_complete=True)
+        print('Sparse scores retrieved')
+
+        ## Getting paper/author data
+
+        submissions_a = self.client.get_all_notes(invitation=self.get_submission_id())
+
+        if same_venue:
+            submissions_b = submissions_a
+            all_submissions = submissions_a
+        else:
+            submissions_b = self.client.get_all_notes(invitation=alternate_venue.get_submission_id())
+            all_submissions = submissions_a + submissions_b
+
+        print(f'{short_name_a}: Retrieved {len(submissions_a)} submissions')
+        print(f'{short_name_b}: Retrieved {len(submissions_b)} submissions')
+
+        papers_by_id_a = {s.id: s for s in submissions_a}
+        papers_by_id_b = papers_by_id_a if same_venue else {s.id: s for s in submissions_b}
+
+        all_authors = set()
+
+        for s in all_submissions:
+            all_authors.update(s.content['authorids']['value'])
+
+        author_profile_by_id = openreview.tools.get_profiles(self.client, all_authors, as_dict=True)
```

Comment:
You can delay the get_profiles call until you really needed it an it seems that you need it when you write the output file so I would move the call before that.

Also it means that you don't need to get the profiles for ALL the submissions, you just need the profiles for the submissions that are "detected" to be possible dual submissions.

Reply:
it looks like for get_all_notes is the same, you just need to get them when you need to output the file.